src/material.cpp

```diff
@@ -93,22 +93,21 @@ void material_type::load( JsonObject &jsobj, const std::string & )
         _dmg_adj.push_back( jsarr.next_string() );
     }
 
-    JsonArray burn_data_array = jsobj.get_array( "burn_data" );
-    for( size_t intensity = 0; intensity < MAX_FIELD_DENSITY; intensity++ ) {
-        if( burn_data_array.has_more() ) {
+    if( jsobj.has_array( "burn_data" ) ) {
+        JsonArray burn_data_array = jsobj.get_array( "burn_data" );
+        while( burn_data_array.has_more() ) {
             JsonObject brn = burn_data_array.next_object();
-            _burn_data[ intensity ] = load_mat_burn_data( brn );
-        } else {
-            // If not specified, supply default
-            bool flammable = _fire_resist <= static_cast<int>( intensity );
+            _burn_data.emplace_back( load_mat_burn_data( brn ) );
+        }
+    } else {
+        // If not specified, supply default
+        if( _fire_resist <= 0 ) {
             mat_burn_data mbd;
-            if( flammable ) {
-                mbd.burn = 1;
-            }
-
-            _burn_data[ intensity ] = mbd;
+            mbd.burn = 1;
+            _burn_data.emplace_back( mbd );
```

Comment:
This should fix the segfault

```suggestion
            mat_burn_data mbd;
            if( _fire_resist <= 0 ) {
                    mbd.burn = 1;
            }
            _burn_data.emplace_back( mbd );
```